src/Units/UnitState.java

```diff
@@ -87,64 +87,62 @@ public boolean hasMaterials()
 
   public boolean isHurt()
   {
-    return health < healthFromHP(UnitModel.MAXIMUM_HP);
+    return getHP() < UnitModel.MAXIMUM_HP;
   }
   public int getHP()
   {
-    return (int) Math.ceil(healthToHP(health));
+    return roundHP(health);
   }
-  /** @return value in range [0-1.0]; represents the unit's current effectiveness */
-  public double getHPFactor()
+  public int getHPFactor()
   {
-    return getHP() / (double) UnitModel.MAXIMUM_HP;
+    return roundHP(health) / 10;
   }
-  /** @return un-rounded HP */
-  public double getPreciseHP()
+  public static int roundHP(int health)
   {
-    return healthToHP(health);
-  }
-  public static double healthToHP(int input)
-  {
-    return ((double) input) / 10;
-  }
-  public static int healthFromHP(double input)
-  {
-    return (int) (input * 10);
+    if( health >= 0 )
+      //     "round up", then kill the last digit
+      return (health + 9) / 10 * 10;
+    // Truncation rounds toward zero, so we need to round down for negative values
+    return (health - 9) / 10 * 10;
```

Comment:
Is "units with negative health" a common use case?

Reply:
I didn't think so either, but then I remembered we use `UnitContext`->`UnitState` as part of `UnitDelta` to store combat damage.
If this didn't work right, we'd potentially get pretty weird results from simulated damage calcs (which allow overkill)